pkg/alicloud/client/client_dns.go

```diff
@@ -119,6 +132,47 @@ func (d *dnsClient) DeleteDomainRecords(ctx context.Context, domainName, name, r
 	return nil
 }
 
+func (d *dnsClient) getDomainsWithCache() (map[string]alidns.Domain, error) {
+	// cache.Expiring Get and Set methods are concurrency-safe
+	// However, if an accessKeyID is not present in the cache and multiple DNSRecords are reconciled at the same time,
+	// it may happen that getDomains is called multiple times instead of just one, so use a mutex to guard against this
+	d.domainsCacheMutex.Lock()
```

Comment:
This is a global lock. Not sure in the normal case, how many accessKeyIDs will be served for a single extension.
It is not a good practice as  `	domains, err := d.getDomains()` is a long call.

Reply:
In the worst case, as many accessKeyIDs as we have shoots on a single seed (assuming every shoot uses a different accessKeyID, which is very unlikely), and the max number of such shoots is 250 currently (can be configured, but this is the default value, and for many reasons it should not be exceeded by much). In reality, we should assume no more than ~100 accessKeyIDs. But getting all of them to contend for the same lock is very unlikely, since we would need to have `DNSRecords` resources of many different shoots being reconciled at the same time. There are precautions to prevent too many shoots from being reconciled at once (e.g. configurable jitter in `gardenlet`), so I would expect `DSNRecords` of no more than 10-20 different shoots being reconciled at the same time, and it's likely that at least for some of these the domains will be already present in the cache.

I get your point now that it's possible to avoid this by introducing somewhat more complex locking, but I would rather not make it more complex than needed for now.

Reply:
I also checked current shoots for AliCloud - only one custom domain exists. Current code is OK. Would you please add some comment for this lock to depict the potential issue if more custom domains emerge than expected?

Reply:
OK, I added such a comment.